### ISSUE TITLE
on_duplicate_key_update ignoring passed-in updated_at time, using current server time

### ISSUE THREAD

**Issue by Rob117** [zdennis/activerecord-import#565]
Essentially, I have code that does something like the following:

```
results = Books.import books_for_import, on_duplicate_key_update: {
      conflict_target: %i[client_id user_id], columns: %i[updated_at book_text
                                                          book_note book_title]
    }
```

The problem is that the `updated_at` times that I pass in are being ignored for automatically generated ones. If I change it to `timestamps: false`, then I can't pass in new objects because `created_at ` won't be populated even if the current value is nil. Is there some better way to handle this so that I can use the updated_at times that I'm passing in, but still have created_at be generated if it's false?

The only other work around I can think of is to get all potentially existing objects from the database, merge their created at times, and then put them back, but that is a very expensive operation that defeats the point of import in the first place.

Any and all feedback appreciated, and thank you for the gem.

**Comment by Rob117** [comment]
Also, odd things are happening with timestamps: false being set. Quick peek at rails code:

```
results = Flashcard.import cards_for_import, on_duplicate_key_update: {
      conflict_target: %i[client_id user_id], columns: %i[created_at updated_at deck_id keyword_text
                                                          keyword_meaning example_usage keyword_locale
                                                          front_note back_note], timestamps: false
    }
```

Note that timestamps are marked to false. However, when I pass in a hash, this is what happens:

raw hash before import:

```
<Flashcard id: nil, user_id: 1, deck_id: 1542622016669, keyword_text: "The first of many", keyword_meaning: "una conturbo avoco aut tondeo", example_usage: "Autem sunt ut. Sit ipsam minima. Earum libero accu...", keyword_locale: "38hr5bvo2b", front_note: "Ex praesentium eos veniam.", back_note: "Est in deleniti quisquam.", created_at: "2018-11-19 10:05:56", updated_at: "2018-11-19 10:05:57", deleted_at: nil, client_id: 1542622136682>
```
`updated_at`, but in iso8601(6) - `"2018-11-19T10:05:56.670119Z"`

wait a bit, step past import line with debugger, view same hash:
```
<Flashcard id: 1, user_id: 1, deck_id: 1542622016669, keyword_text: "The first of many", keyword_meaning: "una conturbo avoco aut tondeo", example_usage: "Autem sunt ut. Sit ipsam minima. Earum libero accu...", keyword_locale: "38hr5bvo2b", front_note: "Ex praesentium eos veniam.", back_note: "Est in deleniti quisquam.", created_at: "2018-11-19 10:06:18", updated_at: "2018-11-19 10:06:18", deleted_at: nil, client_id: 1542622136682>
```

Important part! `updated_at` in iso8601(6) = `"2018-11-19T10:06:18.085401Z"`

Unless I misread the docs, specifying timestamps: false should be preventing this overwrite. The time in the database is using the second, later time as well - not the first time that I specified and passed to the column.

This looks like a serious bug, but if I'm using it wrong somehow please let me know and I'll gladly amend the documentation.

Thank you!

**Comment by Rob117** [comment]
Update!

I think if you move `timestamps: false` to BEFORE the `on_duplicate_key_update:` it appears to work!

**Comment by jkowens** [comment]
Yes, it looks like in your previous example the `timestamps` param had been included inside the `on_duplicate_key_update` hash. This should work too:

```
results = Flashcard.import cards_for_import, on_duplicate_key_update: {
      conflict_target: %i[client_id user_id], columns:
         %i[created_at updated_at deck_id keyword_text
         keyword_meaning example_usage keyword_locale
         front_note back_note]
    }, timestamps: false
```

**Comment by jkowens** [comment]
I will look at what we can do to allow `updated_at` to be manually set. Thanks for reporting this issue.

**Comment by Rob117** [comment]
@jkowens Thank you!

**Comment by Rob117** [comment]
> Yes, it looks like in your previous example the `timestamps` param had been included inside the `on_duplicate_key_update` hash. This should work too:
> 
> ```
> results = Flashcard.import cards_for_import, on_duplicate_key_update: {
>       conflict_target: %i[client_id user_id], columns:
>          %i[created_at updated_at deck_id keyword_text
>          keyword_meaning example_usage keyword_locale
>          front_note back_note]
>     }, timestamps: false
> ```

I can't believe I missed this, thank you so much. I appreciate that.

**Comment by jkowens** [comment]
Ability to manually set updated timestamps released in v0.28.0 👍
